lib/finicity/v1/request/generate_finicity_connect_link.rb

```diff
@@ -27,10 +28,19 @@ def body
         data['customerId'] = "#{customer_id}"
         data['redirectUri'] = redirect_uri
         data['type'] = 'aggregation'
-        data['oauthOptions'] = {
-          'enabled' => oauth_support,
-          'autoReplace' => autoreplace
-        }
+        if oauth_support
+          if autoreplace
+            data['oauthOptions'] = {
+              'enabled' => oauth_support,
+              'autoReplace' => autoreplace
+            }
+          else
+            data['oauthOptions'] = {
+              'enabled' => oauth_support,
+              'institutions' => institutions
+            }
+          end
+        end
         data.to_json(:root => 'data')
```

Comment:
This is not part of the diff, but I'm wondering what the `:root => 'data'` is for? Looking at documentations, I think `root` is supposed to be a boolean parameter. In this case I don't think it does anything.